libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -83,37 +84,34 @@ internal class MapboxDirectionsSession(
      * Fetch route based on [RouteOptions]
      *
      * @param routeOptions RouteOptions
-     * @param routesRequestCallback Callback that gets notified with the results of the request(optional),
+     * @param routerCallback Callback that gets notified with the results of the request(optional),
      * see [registerRoutesObserver]
      *
      * @return requestID, see [cancelRouteRequest]
      */
     override fun requestRoutes(
         routeOptions: RouteOptions,
-        routesRequestCallback: RoutesRequestCallback
+        routerCallback: RouterCallback
     ): Long {
         return router.getRoute(
             routeOptions,
-            object : Router.Callback {
-                override fun onResponse(routes: List<DirectionsRoute>) {
-                    routesRequestCallback.onRoutesReady(routes)
+            object : RouterCallback {
+                override fun onRoutesReady(routes: List<DirectionsRoute>) {
+                    routerCallback.onRoutesReady(routes)
                 }
 
-                override fun onFailure(throwable: Throwable) {
-                    logger.e(
-                        Tag(TAG),
-                        Message(
-                            """
-                                Route request - failure: $throwable
-                                For options: $routeOptions
-                            """.trimIndent()
+                override fun onFailure(reasons: List<RouterFailure>, routeOptions: RouteOptions) {
+                    reasons.forEach {
+                        LoggerProvider.logger.e(
+                            Tag(TAG),
+                            Message(it.toString())
                         )
```

Comment:
Do we really want to log out now that the `reasons` are available? 🤔 

Same thing applies to `onCanceled`.

Reply:
Probably it's not necessary anymore :+1:

Reply:
Alternatively, we can decrease the log level to keep this around by default but not pollute the top log levels as much.

Reply:
Opted to remove.